clients/snowflake/streaming.go

```diff
@@ -26,6 +27,45 @@ import (
 // https://docs.snowflake.com/en/user-guide/snowpipe-streaming/snowpipe-streaming-high-performance-limitations#channel-limits
 const maxChunkSize = 4 * 1024 * 1024 // 4MB
 
+// Error codes that require channel reopening
+// https://docs.snowflake.com/en/user-guide/snowpipe-streaming/snowpipe-streaming-high-performance-error-handling#channel-errors-that-require-a-manual-reopen
+var channelReopenErrorCodes = map[string]bool{
+	"ERR_CHANNEL_HAS_INVALID_ROW_SEQUENCER":           true,
+	"ERR_CHANNEL_HAS_INVALID_CLIENT_SEQUENCER":        true,
+	"ERR_CHANNEL_MUST_BE_REOPENED":                    true,
+	"ERR_CHANNEL_MUST_BE_REOPENED_DUE_TO_ROW_SEQ_GAP": true,
+	"ERR_CHANNEL_DOES_NOT_EXIST_OR_IS_NOT_AUTHORIZED": true,
+	"STALE_PIPE_CACHE":                                true,
+}
+
+// ChannelReopenableError represents an error that can be resolved by reopening the channel
+type ChannelReopenableError struct {
```

Comment:
Hm, can we make this more generic?

```go
// ErrorResponse - Example of an error from Snowflake --
// {"code":"392604","message":"Request execution in progress. Use provided Location header or result handler id to perform query monitoring and management.","result_handler":"01bdd64c-0616-7b86-0066-c80327f556221"}
type ErrorResponse struct {
	Code          string `json:"code"`
	Message       string `json:"message"`
	ResultHandler string `json:"result_handler"`
}

func (e ErrorResponse) Format() string {
	return fmt.Sprintf("%s (code: %s)", e.Message, e.Code)
}
```

Then just add a method on `ErrorResponse` that says `func (e ErrorResponse) ChannelReopenError() bool`